python/understack-workflows/understack_workflows/bmc_password_standard.py

```diff
@@ -0,0 +1,32 @@
+from hashlib import sha256
+from base64 import b64encode
+import re
+
+PASSWORD_LENGTH = 24
+IPV4_REGEXP = r"^\d+\.\d+\.\d+\.\d+$"
+
+def standard_password(ip_addr: str, master_key: str) -> str:
+    """Return a password string for the given IP address
+
+    >>> standard_password("10.3.2.30", "ultra-secret string")
+    'DdushPf7IUTs6oya3ADS0OM3'
+    """
+    if not re.match(IPV4_REGEXP, ip_addr):
+        raise ValueError(f"Needed an IPv4 address, not '{ip_addr}'")
```

Comment:
python has built-in `ipaddress` module standard library since 3.4

```python
import ipaddress
try:
    ip = ipaddress.ip_address(ip_addr)
except ValueError:
    raise ValueError(f"Needed an IPv4 address, not '{ip_addr}'")
```

Might be better than regexp validation

Reply:
Fixed in rebase.  I also corrected the tests.